input/v1beta1/input.go

```diff
@@ -18,7 +18,9 @@ import (
 type GoTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
+	// Go Template Config
+	// +optional
+	Config *Config `json:"config,omitempty"`
```

Comment:
maybe config is a bit generic? all the other fields below (source/inline/filesystem) could easily be below `config`. WDYT about `templateConfig`? CC @turkenh